partiql-parser/src/parse/mod.rs

```diff
@@ -378,7 +380,42 @@ mod tests {
             )
             AS deltas FROM SOURCE_VIEW_DELTA_FULL_TRANSACTIONS delta_full_transactions
             "#;
-            parse!(q)
+            parse!(q);
+        }
+    }
+
+    mod set_ops {
+        use super::*;
+
+        #[test]
+        fn set_ops() {
+            parse!(
+                r#"(SELECT * FROM a LIMIT 10 OFFSET 2) UNION SELECT * FROM b INTERSECT c EXCEPT SELECT * FROM d"#
+            );
+        }
+
+        #[test]
+        fn union_prec() {
+            let l = parse!(r#"a union b union c"#);
+            let r = parse!(r#"(a union b) union c"#);
+            assert_eq!(l, r);
+        }
+
+        #[test]
+        fn intersec_prec() {
+            let l = parse!(r#"a union b intersect c"#);
+            let r = parse!(r#"a union (b intersect c)"#);
+            assert_eq!(l, r);
+        }
+
+        #[test]
+        fn limit() {
+            let l = parse!(r#"SELECT a FROM b UNION SELECT x FROM y LIMIT 10"#);
+            let r = parse!(r#"(SELECT a FROM b UNION SELECT x FROM y) LIMIT 10"#);
+            assert_eq!(l, r);
+            let r2 = parse!(r#"SELECT a FROM b UNION (SELECT x FROM y LIMIT 10)"#);
+            assert_ne!(l, r2);
+            assert_ne!(r, r2);
```

Comment:
nit: could also include tests for `ORDER BY`, `OFFSET`, and a combination of `ORDER BY`, `LIMIT`, and `OFFSET` using the set ops